sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -236,8 +236,10 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
         def createSortMergeJoin() = {
           if (RowOrdering.isOrderable(leftKeys)) {
-            Some(Seq(joins.SortMergeJoinExec(
-              leftKeys, rightKeys, joinType, nonEquiCond, planLater(left), planLater(right))))
+            val (streamed, buffered) = getSortMergeJoinStreamedAndBufferedSide(joinType, leftKeys,
+              left, right, conf)
+            Some(Seq(joins.SortMergeJoinExec(leftKeys, rightKeys, joinType, nonEquiCond,
```

Comment:
Just curious: does the order of output attributes matter?

https://github.com/apache/spark/blob/d8920ae13346670bac3f1a7e49dfc2ba85c93b03/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledJoin.scala#L62

iiuc, BHJ has an BuildSide enum and doesn't actually switch left/right children so it doesn't have this problem?

Reply:
yeah, I think we should change it back to the original smj output ordering after we switch join side.